core/src/test/java/mediumtest/issues/ShowIssueMediumTests.java

```diff
@@ -0,0 +1,184 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+class ShowIssueMediumTests {
+
+  private static final String ISSUE_KEY = "myIssueKey";
+  private static final String PROJECT_KEY = "projectKey";
+  private static final String CONNECTION_ID = "connectionId";
+  private static final String CONFIG_SCOPE_ID = "configScopeId";
+  public static final String RULE_KEY = "ruleKey";
+  private ServerFixture.Server serverWithIssue = newSonarQubeServer("10.2")
+    .withProject(PROJECT_KEY,
+      project -> project.withBranch("branchName",
+        branch -> branch.withIssue(ISSUE_KEY, RULE_KEY, "msg", "author", "file/path", "OPEN", "", "2023-05-13T17:55:39+0202",
+          new TextRange(1, 0, 3, 4))))
+    .withSourceFile("projectKey:file/path", sourceFile -> sourceFile.withCode("source\ncode\nfile"))
+    .start();
+  private SonarLintTestBackend backend;
+
+  @AfterEach
+  void tearDown() throws ExecutionException, InterruptedException {
+    backend.shutdown().get();
+    if (serverWithIssue != null) {
+      serverWithIssue.shutdown();
+      serverWithIssue = null;
+    }
+  }
+
+  @Test
+  void it_should_update_the_telemetry_on_show_issue() {
+    var fakeClient = newFakeClient().build();
+    backend = newBackend()
+      .withSonarQubeConnection(CONNECTION_ID, serverWithIssue)
+      .withBoundConfigScope(CONFIG_SCOPE_ID, CONNECTION_ID, PROJECT_KEY)
+      .withEmbeddedServer()
+      .build(fakeClient);
+
+    assertThat(backend.telemetryFilePath())
+      .content().asBase64Decoded().asString()
+      .contains("\"showIssueRequestsCount\":0");
+
+    var showIssueRequestHandler = new ShowIssueRequestHandler(newFakeClient().build(), backend.getConnectionConfigurationRepository(),
+      backend.getConfigurationServiceImpl(), backend.getBindingSuggestionProviderImpl(), backend.getServerApiProvider(),
+      backend.getTelemetryServiceImpl());
+
+    showIssueRequestHandler.showIssue(new ShowIssueRequestHandler.ShowIssueQuery(serverWithIssue.baseUrl(), "projectKey", "issueKey"));
+
+    assertThat(backend.telemetryFilePath())
+      .content().asBase64Decoded().asString()
+      .contains("\"showIssueRequestsCount\":1");
+  }
+
+  @Test
+  void it_should_open_an_issue_in_ide() throws Exception{
+    var issueKey = "myIssueKey";
+    var projectKey = "projectKey";
+    var connectionId = "connectionId";
+    var configScopeId = "configScopeId";
+
+    var fakeClient = newFakeClient().build();
+    backend = newBackend()
+      .withSonarQubeConnection(connectionId, serverWithIssue)
+      .withBoundConfigScope(configScopeId, connectionId, projectKey)
+      .withEmbeddedServer()
+      .build(fakeClient);
+
+    HttpRequest request = openIssueWithProjectAndKeyRequest("&issue=" + issueKey, "&project=" + projectKey);
+
+    var response = java.net.http.HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString());
```

Comment:
You could maybe extract those 2 lines in a method to improve the readability of the tests